tests/testthat/test-gcov.R

```diff
@@ -1,48 +1,68 @@
 test_that("parse_gcov parses files properly", {
-  mockery::stub(parse_gcov, "file.exists", TRUE)
-  mockery::stub(normalize_path, "normalizePath", "simple.c")
-  mockery::stub(parse_gcov, "line_coverages", function(source_file, matches, values, ...) values)
+  local_mocked_bindings(
+    # Only called within parse_gcov
+    source_file_exists = function(path) TRUE,
+    # Only called within normalize_path
+    normalize_path_impl = function(path) "simple.c",
+    # Only called within parse_gcov
+    line_coverages_impl = function(source_file, matches, values, ...) values
+  )
 
+  with_mocked_bindings(
+    expect_equal(parse_gcov("hi.c.gcov"), numeric()),
+    read_lines_impl = function(x) {
+      "        -:    0:Source:simple.c"
+    }
+  )
 
-  mockery::stub(parse_gcov, "readLines",
-"        -:    0:Source:simple.c")
-  expect_equal(parse_gcov("hi.c.gcov"), numeric())
```

Comment:
Inside `parse_gcov()`, I replaced `readLines()` by `read_lines_impl()` to be able to mock `readLines()` only inside `parse_gcov()`